qiskit_nature/second_q/problems/eigenstate_result.py

```diff
@@ -14,166 +14,121 @@
             return energies[0].real
         return None
 
     @property
-    def groundstate(
-        self,
-    ) -> Optional[
-        Union[
-            str,
-            dict,
-            Result,
-            list,
-            np.ndarray,
-            Statevector,
-            QuantumCircuit,
-            Instruction,
-            OperatorBase,
-        ]
-    ]:
-        """returns ground state"""
+    def groundstate(self) -> QuantumCircuit | None:
+        """TODO."""
         states = self.eigenstates
         if states:
             return states[0]
         return None
 
-    @property
-    def aux_operator_eigenvalues(self) -> Optional[List[dict[str, Tuple[complex, complex]]]]:
-        """return aux operator eigen values"""
-        return self._aux_operator_eigenvalues
+    @classmethod
+    def from_result(
+        cls, raw_result: EigenstateResult | EigensolverResult | MinimumEigensolverResult
+    ) -> EigenstateResult:
+        """TODO."""
+        if isinstance(raw_result, EigenstateResult):
+            return raw_result
+        if isinstance(raw_result, EigensolverResult):
+            return EigenstateResult.from_eigensolver_result(raw_result)
+        if isinstance(raw_result, MinimumEigensolverResult):
+            return EigenstateResult.from_minimum_eigensolver_result(raw_result)
+        raise TypeError("TODO.")
+
+    @classmethod
+    def from_eigensolver_result(cls, raw_result: EigensolverResult) -> EigenstateResult:
+        """TODO."""
+        result = EigenstateResult()
+        result.raw_result = raw_result
+        result.eigenvalues = np.asarray(raw_result.eigenvalues)
+
+        if hasattr(raw_result, "eigenstates"):
+            result.eigenstates = [
+                (_statevector_to_circuit(Statevector(state)), None)
+                for state in raw_result.eigenstates
+            ]
+        elif hasattr(raw_result, "optimal_circuits") and hasattr(raw_result, "optimal_parameters"):
+            result.eigenstates = [
+                (circuit, list(param.values()))
+                for circuit, param in zip(
+                    raw_result.optimal_circuits, raw_result.optimal_parameters
+                )
+            ]
 
-    @aux_operator_eigenvalues.setter
-    def aux_operator_eigenvalues(self, value: List[dict[str, Tuple[complex, complex]]]) -> None:
-        """set aux operator eigen values"""
-        self._aux_operator_eigenvalues = value
+        if raw_result.aux_operators_evaluated is not None:
+            result.aux_operators_evaluated = []
+            for aux_op_eval in raw_result.aux_operators_evaluated:
+                aux_op_values: ListOrDict[complex]
+                if isinstance(aux_op_eval, list):
+                    aux_op_values = [val[0] for val in aux_op_eval]
+                else:
+                    aux_op_values = {key: val[0] for key, val in aux_op_eval.items()}
+                result.aux_operators_evaluated.append(aux_op_values)
```

Comment:
Note to self: extract this into a utility method to remove code duplication with below